src/uct/tcp/tcp_sockcm.c

```diff
@@ -57,13 +57,26 @@ static ucs_status_t uct_tcp_sockcm_event_err_to_ucs_err_log(int fd,
     ucs_status_t status;
 
     status = ucs_socket_getopt(fd, SOL_SOCKET, SO_ERROR, (void*)&error, sizeof(error));
-    if ((status != UCS_OK) || (error != ECONNREFUSED)) {
-        *log_level = UCS_LOG_LEVEL_ERROR;
-        return UCS_ERR_IO_ERROR;
+    if (status != UCS_OK) {
+        goto err;
     }
 
-    *log_level = UCS_LOG_LEVEL_DEBUG;
-    return UCS_ERR_REJECTED;
+    ucs_debug("error event on fd %d: %s", fd, strerror(error));
+
+    switch (error) {
+    /* UCS_ERR_REJECT is returned only for user's explicit reject */
```

Comment:
IIUC, ECONNREFUSED should be handled as reject or conn_reset, unreachable is not correct.

Reply:
we said that ucs_err_reject is set only for user's reject. network reject is passed to upper layer as unreachable.

Reply:
if server rejects due to SOMAXCONN overflow, that's not a network reject

Reply:
you think it should be considered as user's reject?

Reply:
not "user's" but yes - it's kind of reject

Reply:
@yosefe what do you think? 
rdmacm distinguishes between user's reject and network reject with the priv_data. 
if the client gets a reject_event, without priv_data, how would it know if it's due to SOMAXCONN or another network reject?

Reply:
we should return UCS_ERR_NOT_CONNECTED or UCS_ERR_CONNECTION_RESET
from API perspective it's not a reject because reject wsa not called on server side

Reply:
ok, will 'translate' ECONNREFUSED on tcp_sockcm to UCS_ERR_NOT_CONNECTED for connect_cb in ucp.
should we attempt to fallback to next sockaddr tl in this case?